Login and roster upload

jrm86 opened the pull request:
Implementation of the Login and Roster Upload functionality.

Includes the authentication stub written by @afig to facilitate login.

jrm86 committed:
initial_changes

Mostly draft work.  Still needs some editing/testing.

jrm86 committed:
changes to login functionality

jrm86 committed:
expand_collapse roster upload

Made roster upload a collapsible section of the UI (default hidden)

Added function to change the arrow icon when expanding/collapsing upload section

jrm86 committed:
server update for login

DasilvaB commented:
In index.js on line 138 you have:

`reader.onload = uploadRoster;`

I'm assuming you meant to call the method, so would need the () and to pass in the event? Also inside the method uploadRoster on line 145 its missing a space.

jrm86 committed:
moved block of code 

Moved a block of code so that roster upload would appear on Login page only, and not on the "About" page.

afig commented on src/webapp/client/js/index.js:
```diff
@@ -121,8 +129,25 @@ $(document).ready(function() {
+
+	$('#uploadBtn').click(function()
+	{
+		var file = document.getElementById('#rosterImport').files[0];
```
```suggestion
		dbInfo = getDBFields();
		var file = document.getElementById('rosterImport').files[0];
```

afig replied:
The `#` is not necessary when using `document.getElementByID()` (only when using `$()`)

Also, the database info need to also be send with the uploaded file, so this is part 1 of transmitting that info.

afig commented on src/webapp/client/js/index.js:
```diff
@@ -121,8 +129,25 @@ $(document).ready(function() {
 
+function uploadRoster(event){
+	var result = event.target.result;
+    var fileName = document.getElementById('#rosterImport').files[0].name;
```
```suggestion
	var fileName = document.getElementById('rosterImport').files[0].name;
```

afig replied:
Changed spaces to tab and removed unnecessary `#`.

afig commented on src/webapp/client/js/index.js:
```diff
@@ -121,8 +129,25 @@ $(document).ready(function() {
+	var result = event.target.result;
+    var fileName = document.getElementById('#rosterImport').files[0].name;
+	
+	$.post('/importSectionRoster', { data: result, name: fileName }, showAlert("<p>Upload Successful.</p>"));
```
```suggestion
	var data = $.extend({}, dbInfo, {data: result}, {name: fileName});
	$.post('/importSectionRoster', data, showAlert("<p>Upload Successful.</p>"));
```

afig replied:
As mentioned in an earlier suggestion, the DB info needs to be included in the upload. Here, a new object is being created that includes both the data being uploaded as well as the connection info.

afig reviewed (commented):
Added some suggestions that address issues with file uploads on the client side. There are a few more things that needed to be addressed on the middleware side to allow uploads to work correctly. I can discuss those tomorrow.

afig committed:
Update src/webapp/client/js/index.js

Co-Authored-By: jrm86 <jrm86git@gmail.com>

afig committed:
Update src/webapp/client/js/index.js

Co-Authored-By: jrm86 <jrm86git@gmail.com>

afig committed:
Update src/webapp/client/js/index.js

Co-Authored-By: jrm86 <jrm86git@gmail.com>

jrm86 commented:
Thank you for the suggestions and fixes @afig.  Had some issues with my local install of the DB portion, which made testing a little troublesome.  We'll discuss this and other changes during today's meeting.

afig committed:
Merge branch 'dev' into login_and_rosterUpload

afig reviewed (approved):
Looks good to merge. I added a commit that addressed the merge conflict.

DasilvaB reviewed (approved):
Nice job, looks good.
